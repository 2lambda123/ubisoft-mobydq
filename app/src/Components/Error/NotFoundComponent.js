import React from 'react';
import {styles} from './../../styles/baseStyles';
import {withStyles} from '@material-ui/core/styles';

class NotFoundComponent extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <React.Fragment>
                <br/><br/><br/>
                <h1>404 Not Found</h1>
            </React.Fragment>
        )
    }
}

export default withStyles(styles)(NotFoundComponent);
