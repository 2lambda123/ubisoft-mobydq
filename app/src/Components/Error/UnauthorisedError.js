import React from 'react';
import './../../styles/baseStyles';
import LinkButton from './../FormInput/LinkButton';

export const UnauthorisedError = () => <div style={{ 'display': 'flex', 'justifyContent': 'center', 'alignItems': 'center' }}>
  <div>
    <h4>Login expired</h4>
    <LinkButton disabled={false} label="Click here to re-login" type="Create" color="primary"
      variant="contained" to={'/login'}/>
  </div>
</div>;
