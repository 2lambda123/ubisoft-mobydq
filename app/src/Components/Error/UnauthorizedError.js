import React from 'react';
import './../../styles/baseStyles';
import LinkButton from './../FormInput/LinkButton';

export const UnauthorizedError = () => <div style={{ 'display': 'flex', 'justifyContent': 'center', 'alignItems': 'center' }}>
  <div>
    <div style={{ 'textAlign': 'center' }}>
      Your session has expired
    </div>
    <div style={{ 'textAlign': 'center' }}>
      <LinkButton label="Re-login" type="Create" color="secondary" variant="contained" to={'/login'}/>
    </div>
  </div>
</div>;
