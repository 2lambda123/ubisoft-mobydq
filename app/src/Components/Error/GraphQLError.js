import React from 'react';
import { UnauthorisedError } from './UnauthorisedError';

class GraphQLError extends React.Component {
  render() {
    const error = this.props.error;
    if (error.graphQLErrors.length > 0) {
      return <div>
        {error.graphQLErrors.statusCode + ': ' + error.graphQLErrors.result.message}
      </div>;
    }
    if (error.networkError) {
      switch (error.networkError.statusCode) {
        case 401:
          return <UnauthorisedError/>;
        default:
          return <div>
            {error.networkError.statusCode + ': ' + error.networkError.result.message}
          </div>;
      }
    }
    return <div>No error. Something wrong happened...</div>;
  }
}

export default GraphQLError;
