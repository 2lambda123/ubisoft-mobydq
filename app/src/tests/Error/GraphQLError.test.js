import React from 'react';
import { shallowWrap } from './../../setupTests';

import { MemoryRouter } from 'react-router-dom';

import { GraphQLError } from '../../Components/Error/GraphQLError';

describe('graphql error unit test', () => {
  const gqlError = {
    'graphQLErrors': [
      {
        'message': 'test message 1'
      },
      {
        'message': 'test message 2'
      }
    ],
    'message': 'a global error message'
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallowWrap(
      <GraphQLError
        error={gqlError}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('network error unit test', () => {
  const networkError = {
    'networkError': {
      'statusCode': 401,
      'message': 'Unauthorised'
    },
    'graphQLErrors': []
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallowWrap(
      <GraphQLError
        error={networkError}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('error component functional test', () => {

  it('renders the graphql error', () => {
    const gqlError = {
      'graphQLErrors': [
        {
          'error': 'error message 1'
        },
        {
          'error': 'error message 2'
        }
      ],
      'message': 'a global error message'
    };
    const wrapper = mount(<div><GraphQLError error={gqlError}/></div>);
    const errorMessage = <div>a global error message</div>;
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });

  it('renders the generic network error', () => {
    const networkError = {
    'networkError': {
      'statusCode': 404,
      'message': 'Not found'
    },
    'graphQLErrors': []
  };
    const wrapper = mount(<div><GraphQLError error={networkError}/></div>);
    const errorMessage = <div>404: Not found</div>;
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });
});


describe('error component functional test', () => {

  it('renders the graphql error', () => {
    const gqlError = {
      'graphQLErrors': [
        {
          'error': 'error message 1'
        },
        {
          'error': 'error message 2'
        }
      ],
      'message': 'a global error message'
    };
    const wrapper = mount(<div><GraphQLError error={gqlError}/></div>);
    const errorMessage = <div>a global error message</div>;
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });

  it('renders the generic network error', () => {
    const networkError = {
    'networkError': {
      'statusCode': 404,
      'message': 'Not found'
    },
    'graphQLErrors': []
  };
    const wrapper = mount(<div><GraphQLError error={networkError}/></div>);
    const errorMessage = <div>404: Not found</div>;
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });

  it('renders the 401 login error', () => {
    const networkError = {
    'networkError': {
      'statusCode': 401,
      'message': 'Unauthorised'
    },
    'graphQLErrors': []
  };
    const wrapper = mount(<div><MemoryRouter><GraphQLError error={networkError}/></MemoryRouter></div>);
    const errorMessage = 'Your session has expired'
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });

  it('renders the unknown error message', () => {
    const networkError = {
    'someUnexpectedJson': 'message'
  };
    const wrapper = mount(<div><MemoryRouter><GraphQLError error={networkError}/></MemoryRouter></div>);
    const errorMessage = <div>No error. Something wrong happened...</div>;
    expect(wrapper.contains(errorMessage)).toEqual(true);
  });
});