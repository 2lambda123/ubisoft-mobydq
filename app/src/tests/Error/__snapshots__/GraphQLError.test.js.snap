// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql error unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GraphQLError
    error={
      Object {
        "graphQLErrors": Array [
          Object {
            "message": "test message 1",
          },
          Object {
            "message": "test message 2",
          },
        ],
        "message": "a global error message",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "a global error message",
    },
    "ref": null,
    "rendered": "a global error message",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "a global error message",
      },
      "ref": null,
      "rendered": "a global error message",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;

exports[`network error unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GraphQLError
    error={
      Object {
        "graphQLErrors": Array [],
        "networkError": Object {
          "message": "Unauthorised",
          "statusCode": 401,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;
